tests/unit/lms/services/canvas_test.py

```diff
@@ -1,89 +1,282 @@
+        self, finder
+    ):
+        finder.assert_file_in_course(sentinel.course_id, "2")
+
+    def test_assert_file_in_course_raises_if_the_file_isnt_in_the_course(self, finder):
+        with pytest.raises(CanvasFileNotFoundInCourse):
+            finder.assert_file_in_course(sentinel.course_id, "4")
+
+    def test_find_matching_file_in_course_returns_the_matching_file_id(
+        self, finder, canvas_api_client, file_service
     ):
         # The file dict from the Canvas API that we expect the search to match.
         matching_file_dict = canvas_api_client.list_files.return_value[1]
-
-        file_ = factories.File(
+        file_service.get.return_value = factories.File(
             name=matching_file_dict["display_name"],
             size=matching_file_dict["size"],
         )
 
-        matching_file_id = canvas_service.find_matching_file_in_course(
-            sentinel.course_id, file_
+        matching_file_id = finder.find_matching_file_in_course(
+            sentinel.course_id, sentinel.file_id
         )
 
+        file_service.get.assert_called_once_with(sentinel.file_id, type_="canvas_file")
         canvas_api_client.list_files.assert_called_once_with(sentinel.course_id)
         assert matching_file_id == str(matching_file_dict["id"])
 
-    def test_it_returns_None_if_theres_no_matching_file_in_the_course(
-        self, canvas_service
+    def test_find_matching_file_in_course_returns_None_if_theres_file_in_the_db(
+        self, finder, file_service
+    ):
+        file_service.get.return_value = None
+
+        assert not finder.find_matching_file_in_course(
+            sentinel.course_id, sentinel.file_id
+        )
+
+    def test_find_matching_file_in_course_returns_None_if_theres_no_match(
+        self, finder, file_service
+    ):
+        file_service.get.return_value = factories.File(name="foo")
+
+        assert not finder.find_matching_file_in_course(
+            sentinel.course_id, sentinel.file_id
+        )
+
+    def test_find_matching_file_in_course_doesnt_return_the_same_file(
+        self, finder, canvas_api_client, file_service
     ):
-        assert not canvas_service.find_matching_file_in_course(
-            sentinel.course_id, factories.File()
+        # If the response from the Canvas API contains a "matching" file dict
+        # that happens to be the *same* file as the one we're searching for (it
+        # has the same id) find_matching_file_in_course() should not return
+        # the same file_id as it was asked to search for a match for.
+        matching_file_dict = canvas_api_client.list_files.return_value[1]
+        file_service.get.return_value = factories.File(
+            lms_id=str(matching_file_dict["id"]),
+            name=matching_file_dict["display_name"],
+            size=matching_file_dict["size"],
+        )
+
+        assert not finder.find_matching_file_in_course(
+            sentinel.course_id, sentinel.file_id
```

Comment:
You could probably cover this case by extending the fixture in `test_find_matching_file_in_course_returns_the_matching_file_id`

```python
       canvas_api_client.list_files.return_value = [
            {"id": 1, "display_name": "Wrong name", "size": file.size},
            {"id": 2, "display_name": file.name, "size": 99999999999},
            # We should skip this one as it has the same id
            {"id": file.lms_id, "display_name": file.name, "size": file.size},
            {"id": 3, "display_name": file.name, "size": file.size},
        ]
```